sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.cpp

```diff
@@ -12,8 +12,241 @@
 
 #include "context.hpp"
 #include "event.hpp"
+#include "usm.hpp"
+
 #include "ur_level_zero.hpp"
 
+#include <umf_helpers.hpp>
+
+ur_result_t umf2urResult(umf_result_t umfResult) {
+  if (umfResult == UMF_RESULT_SUCCESS)
+    return UR_RESULT_SUCCESS;
+
+  switch (umfResult) {
+  case UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY:
+    return UR_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+  case UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC: {
+    auto hProvider = umfGetLastFailedMemoryProvider();
+    if (hProvider == nullptr)
+      return UR_RESULT_ERROR_UNKNOWN;
+
+    ur_result_t Err = UR_RESULT_ERROR_UNKNOWN;
+    umfMemoryProviderGetLastNativeError(hProvider, nullptr,
+                                        reinterpret_cast<int32_t *>(&Err));
+    return Err;
+  }
+  case UMF_RESULT_ERROR_INVALID_ARGUMENT:
+    return UR_RESULT_ERROR_INVALID_ARGUMENT;
+  case UMF_RESULT_ERROR_INVALID_ALIGNMENT:
+    return UR_RESULT_ERROR_UNSUPPORTED_ALIGNMENT;
+  case UMF_RESULT_ERROR_NOT_SUPPORTED:
+    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  default:
+    return UR_RESULT_ERROR_UNKNOWN;
+  };
+}
+
+usm::DisjointPoolAllConfigs InitializeDisjointPoolConfig() {
+  const char *PoolConfigVal = std::getenv("UR_L0_USM_ALLOCATOR");
+  if (PoolConfigVal == nullptr) {
+    return usm::DisjointPoolAllConfigs();
+  }
+
+  const char *PoolUrTraceVal = std::getenv("UR_L0_USM_ALLOCATOR_TRACE");
+  const char *PoolPiTraceVal =
+      std::getenv("SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR_TRACE");
+  const char *PoolTraceVal = PoolUrTraceVal
+                                 ? PoolUrTraceVal
+                                 : (PoolPiTraceVal ? PoolPiTraceVal : nullptr);
+
+  bool PoolTrace = false;
+  if (PoolTraceVal != nullptr) {
+    PoolTrace = std::atoi(PoolTraceVal);
+  }
+
+  return usm::parseDisjointPoolConfig(PoolConfigVal, PoolTrace);
+}
```

Comment:
we should still support the SYCL variable as well (it's used in tests).

```suggestion
usm::DisjointPoolAllConfigs InitializeDisjointPoolConfig() {
  const char *PoolUrConfigVal = std::getenv("SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR");
  const char *PoolPiConfigVal = std::getenv("UR_L0_USM_ALLOCATOR");
  const char *PoolConfigVal =
      PoolUrConfigVal ? PoolUrConfigVal : PoolPiConfigVal;
  if (PoolConfigVal == nullptr) {
    return usm::DisjointPoolAllConfigs();
  }

  const char *PoolUrTraceVal = std::getenv("UR_L0_USM_ALLOCATOR_TRACE");
  const char *PoolPiTraceVal =
      std::getenv("SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR_TRACE");
  const char *PoolTraceVal = PoolUrTraceVal
                                 ? PoolUrTraceVal
                                 : (PoolPiTraceVal ? PoolPiTraceVal : nullptr);

  bool PoolTrace = false;
  if (PoolTraceVal != nullptr) {
    PoolTrace = std::atoi(PoolTraceVal);
  }

  return usm::parseDisjointPoolConfig(PoolConfigVal, PoolTrace);
}
```